plugins/chromatic_correction.py

```diff
@@ -0,0 +1,346 @@
+    """Get the file location by adding the directory and date time"""
+    dir_path = os.path.join(get_home_folder(), "Documents/Chromatic aberration correction")
+    os.makedirs(dir_path, exist_ok=True)
+    current_date = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    file_location = os.path.join(dir_path, f"{current_date}_{filename}.{ext}")
+    return file_location
+
+
+def update_chromatic_correction(im: DataArray, points_ref: numpy.ndarray, points_im: numpy.ndarray):
+    """
+    Update the transformation correction parameters that maps features given by points_im to set of features given
+    by point_ref.
+    :param points_ref: (x, y) location of features in reference data
+    :param points_im: (x, y) location of features in given data by im
+    :return: (Dict) transformation values of the given data
+    """
+    if numpy.array_equal(points_ref, points_im):
+        # features of same image are being compared
+        # add default values
+        scale = [1, 1]
+        translation = [0, 0]
+        rotation = 0
+        shear = 0
+        correction_dict = {
+            model.MD_PIXEL_SIZE_COR: scale,
+            model.MD_POS_COR: translation,
+            model.MD_ROTATION_COR: rotation,
+            model.MD_SHEAR_COR: shear
+        }
+    else:
+        # Estimate the transformation matrix
+        scale, translation, rotation, shear = estimate_transformation_from_pointset(points_ref, points_im)
+        # Convert translation from pixels to m/physical units
+        pixel_size = im.metadata[model.MD_PIXEL_SIZE]
+        translation = (translation[0] * pixel_size[0], translation[1] * pixel_size[1])
+
+        # To enable chromatic correction, add MD_CHROMATIC_COR  under the filter wheel component of the
+        # microscope file
+        correction_dict = {
+            model.MD_PIXEL_SIZE_COR: [float(scale[0]), float(scale[1])],
+            model.MD_POS_COR: [float(translation[0]), float(translation[1])],
+            model.MD_ROTATION_COR: float(rotation),
+            model.MD_SHEAR_COR: float(shear)
+        }
+
+    logging.debug(f"Chromatic transformation: {correction_dict}")
+
+    # Update metadata
+    im.metadata.update({model.MD_PIXEL_SIZE_COR: correction_dict[model.MD_PIXEL_SIZE_COR]})
+    im.metadata.update({model.MD_POS_COR: correction_dict[model.MD_POS_COR]})
+    im.metadata.update({model.MD_ROTATION_COR: correction_dict[model.MD_ROTATION_COR]})
+    im.metadata.update({model.MD_SHEAR_COR: correction_dict[model.MD_SHEAR_COR]})
+
+    return correction_dict
+
+
+class ChromaticCorrectionPlugin(Plugin):
+    name = "Chromatic Correction"
+    __version__ = "1.0"
+    __author__ = "Karishma Kumar"
+    __license__ = "GPLv2"
+
+    def __init__(self, microscope, main_app):
+        if "cryosecom-localization" in main_app.main_data.tab.choices.values():
+            super().__init__(microscope, main_app)
+        self.addMenu("Data correction/Lateral chromatic correction", self.start_with_thread)
```

Comment:
Spawning a dialog first, like we discussed yesterday. Make use of the `@call_in_wx_main` decorator. Then there is also no use anymore for the other modal you are spawning (the one with the problems), since we can show problems beforehand. Something like this:
![image](https://github.com/user-attachments/assets/64d75ba7-f642-49eb-85dc-58a12f505124)